app/src/main/java/org/thunderdog/challegram/component/chat/ChatHeaderView.java

```diff
@@ -103,15 +101,22 @@ public void setChat (Tdlib tdlib, TdApi.Chat chat, @Nullable ThreadInfo messageT
     }
 
     if (messageThread != null) {
-      setInnerMargins(Screen.dp(56f), Screen.dp(49f));
-      setText(Lang.plural(messageThread.areComments() ? R.string.xComments : R.string.xReplies, messageThread.getSize()), null);
+      setChatPhoto(chat, chat.photo);
+      setShowVerify(tdlib.chatVerified(chat));
+      setShowScam(tdlib.chatScam(chat));
+      setShowFake(tdlib.chatFake(chat));
+      setShowMute(tdlib.chatNeedsMuteIcon(chat));
+      CharSequence subtitle = messageThread.chatHeaderSubtitle(tdlib);
+      setText(messageThread.chatHeaderTitle(tdlib), subtitle);
+      setForcedSubtitle(subtitle);
```

Comment:
(Mentioned in DM, duplicating here just for tracking): `setForcedSubtitle` call here is not balanced when condition `messageThread == null` is `true`, which causes "X comments" to get stuck in cases when `MessagesController` gets reused and it was in comments / thread view.